### PR TITLE
Tweak node pool usage

### PR DESCRIPTION
**Pull request by leej3**
Allocates some stray cleaml resources to the clearml node-pool and targets the forwardauth deployment resource to the general node pool.

### CODE REVIEW

**Review by viniciusdc — COMMENTED** [review]
LGTM, I will present that in the next meeting with others and we can merge once they have a look as well. Thanks @leej3 for the contribuitions!!!

**Comment by leej3** [comment]
Sounds good. Thanks. One thing to consider is whether to target some of the clearml pods using affinity as we have done in the case of forward-auth. On our deployment the clearml nodes use a lot of resources and  my understanding is that the clearml server needs quite a lot of resources (or at least we haven't explored what the minimum resources are). It may be more sensible to run all auxiliary pods somewhere other than the clearml node pool to be more efficient...

File: qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/chart/values.yaml [hunk 1]
```diff
@@ -226,6 +226,8 @@ redis: # configuration from https://github.com/bitnami/charts/blob/master/bitnam
   master:
     name: "{{ .Release.Name }}-redis-master"
     port: 6379
+    nodeSelector:
```
Comment (costrouc) [inline qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/chart/values.yaml:229]: How is the node being labeled "app: clearml"?

Comment (costrouc) [reply qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/chart/values.yaml:229]: Or this an assumption that the node will have this label?

Comment (leej3) [reply qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/chart/values.yaml:229]: > How is the node being labeled "app: clearml"

good point. It's a default value set in the variable.tf file that propagest to the chart's values.yaml (see [here](https://github.com/Quansight/qhub/blob/29f398af3542143751d52252b8a2204cd9c1e5cf/qhub/template/%7B%7B%20cookiecutter.repo_directory%20%7D%7D/infrastructure/modules/kubernetes/services/clearml/main.tf#L8))). 


> Or this an assumption that the node will have this label?

It is a requirement. We manually set this label for our deployment. We are not sure if this is automated when QHub deploys clearml when using the cloud providers (as described [here](https://docs.qhub.dev/en/stable/source/admin_guide/clearml.html?highlight=clearml#create-a-node-group)).

**Comment by leej3** [comment]
Additionally, instead of targeting all clearml pods to the general pool, we want to be able to separately target the services to different node pools. The agent should run on a large node but all other services (including kube-system components) should be targeted to a general pool with nodes with less resources. @viniciusdc and myself will take a look. We'll submit this separately

File: qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/chart/values.yaml (hunk 1 above)
Comment (leej3) [reply qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/chart/values.yaml:229]: that drew our attention to missing redis/mongdb dynamic value setting. fixed now.

**Comment by viniciusdc** [comment]
c.c @viniciusdc

**Comment by costrouc** [comment]
@viniciusdc could you update this PR when you test clearml? I think this may already be fixed.

**Comment by viniciusdc** [comment]
Based on a comparison between the main.tf files from [this PR](qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/clearml/main.tf) and the [current present on Qhub](https://github.com/Quansight/qhub/blob/main/qhub/template/stages/07-kubernetes-services/modules/kubernetes/services/clearml/main.tf), this was not fixed yet. So we need to move these changes to 0.4.0 standards then we can merge this. (after a new review).

As this will be part of #1217, I will move this to the same milestone.

**Comment by viniciusdc** [comment]
I will be closing this due to the changes made for v0.4.0 and #1292. These changes will then be included once a PR for #1281 is merged. Thanks, @leej3 for pointing out the issue.

**Pull request closed without merging**